dts/bindings/iio/adc/nxp,lpc-lpadc.yaml

```diff
@@ -49,6 +49,10 @@ properties:
     required: true
     description: Offset value B to use if CONFIG_LPADC_DO_OFFSET_CALIBRATION is false
 
+  required-regulators:
+    type: phandles
+    description: References to required regulators which must be enabled for LPADC to function
```

Comment:
you should be using `*-supply` named properties instead, also, this must come with the previous commit to preserve bisectability. Does this lpadc require > 1 regulator?

Reply:
This LPADC does not, no. I think the reasoning here is that this property would ideally be kept flexible, as some LPADC instances do expose a VREFH and VREFL pin, which could hypothetically be linked to multiple regulators.

As an aside (beyond the scope of this PR), should we be supporting a way to specify the target voltage in a regulator phandle? This way the user can select the target voltage the LPADC driver should request each regulator be set to. Something like the way we specify pins in GPIO Phandles

Reply:
then maybe use `vrefh-supply` and `vrefl-supply`?